lib/PostQuery.php

```diff
@@ -167,9 +208,144 @@ protected function get_query() {
+		} elseif ( !$this->pagination && $this->_query instanceof WP_Query ) {
+			$this->pagination = new Pagination($prefs, $this->_query);
 		}
 
 		return $this->pagination;
 	}
+
+	/**
+	 * Override data printed by var_dump() and similar. Realizes the collection before
+	 * returning. Due to a PHP bug, this only works in PHP >= 7.4.
+	 *
+	 * @see https://bugs.php.net/bug.php?id=69264
+	 * @internal
+	 */
+	public function __debugInfo() {
+		return [
+			'__posts'     => $this->getArrayCopy(),
+			'_query'      => $this->_query,
+			'found_posts' => $this->found_posts,
+			'pagination'  => $this->pagination,
+			'factory'     => $this->factory,
+			'iterator'    => $this->getIterator(),
+		];
+	}
+
+	/**
+	 * @internal
+	 */
+	public function getArrayCopy() {
+		// Force eager instantiation of Timber\Posts before returning them all in an array.
+		$this->realize();
+		return parent::getArrayCopy();
+	}
+
+	/**
+	 * For better performance, PostQuery does not instantiate `Timber\Post` objects
+	 * at query time. It instantiates each `Timber\Post` only as needed, i.e. while
+	 * iterating or for direct array access (`$coll[$i]`). Since specific `Timber\Post`
+	 * implementations may have expensive `::setup()` operations, this is usually
+	 * what you want, but not always. For example, you may want to force eager
+	 * instantiation to front-load a collection to be cached. To eagerly instantiate
+	 * a lazy collection of objects is to "realize" that collection.
+	 *
+	 * @api
+	 * @example
+	 * ```php
+	 * $lazyPosts = \Timber\Helper::transient('my_posts', function() {
+	 *   return \Timber\Timber::get_posts([
+	 * 		 'post_type' => 'some_post_type',
+	 *   ]);
+	 * }, HOUR_IN_SECONDS);
+	 *
+	 * foreach ($lazyPosts as $post) {
+	 *   // This will incur the performance cost of Post::setup()
+	 * }
+	 *
+	 * // Contrast with:
+	 *
+	 * $eagerPosts = \Timber\Helper::transient('my_posts', function() {
+	 *   $query = \Timber\Timber::get_posts([
+	 * 		 'post_type' => 'some_post_type',
+	 *   ]);
+	 *   // Incur Post::setup() cost up front.
+	 *   return $query->realize();
+	 * }, HOUR_IN_SECONDS);
+	 *
+	 * foreach ($eagerPosts as $post) {
+	 *   // No additional overhead here.
+	 * }
+	 * ```
+	 * @return \Timber\PostQuery the realized PostQuery
+	 */
+	public function realize() : self {
+		if (!$this->realized) {
+			// offsetGet() is where lazy instantiation actually happens.
+			// Since arbitrary array index access may have happened previously,
+			// leverage that to ensure each Post is instantiated exactly once.
+			// We call parent::getArrayCopy() to avoid infinite mutual recursion.
+			foreach (array_keys(parent::getArrayCopy()) as $k) {
+				$this->offsetGet($k);
+			}
+			$this->realized = true;
+		}
+
+		return $this;
+	}
+
+	/**
+	 * @api
+	 * @return array
+	 */
+	public function to_array() : array {
+		return $this->getArrayCopy();
+	}
+
+	/**
+	 * @todo should we deprecate this method in favor of to_array()?
```

Comment:
Good point, are you saying we should keep this then?

Reply:
I’m not sure yet. I think if we’re going to have collections for other object types like Terms, `to_array()` would make more sense, because then we can be consistent across these object types. And I feel like `to_array()` says better what the method actually does. We already did `Timber::get_posts()`, so another call to `get_posts()` feels a bit weird, right?

What does @jarednova mean here?

Reply:
`to_array()` also makes sense to me here versus `get_posts()` — it also has the benefit of being consistent regardless of the object type (term? user? menu item? etc.)